source/opt/ir_context.h

```diff
@@ -264,6 +296,7 @@ class IRContext {
     valid_analyses_ = valid_analyses_ | kAnalysisDecorations;
   }
 
+  uint32_t unique_id_;
```

Comment:
Needs a comment. to describe this, especially how it starts at 0 but is not the next unique ID to be used.  It's one *less* than the next unique id to be used.